This provides so-called "pthread stubs", which permit thread-safe libraries to
not have to explicitely link with libpthread, and just rely on libc providing
e.g. void pthread_mutex_lock/unlock.  However, because of weak ELF symbols
rules, this will only work as long as we set _dl_dynamic_weak to 1 (see
local-dl-dynamic-weak.diff), so the plan is

- once glibc has integrated Hurd's libpthread and thus is able to provide the
pthread stubs itself, this patch can be dropped.
- once we see that all packages got recompiled with that new glibc (i.e.
apt-cache rdepends libpthread-stubs0 is empty), the local-dl-dynamic-weak.diff
patch can be dropped.

Index: Makerules
===================================================================
RCS file: /cvs/glibc/libc/Makerules,v
retrieving revision 1.450
diff -u -p -r1.450 Makerules
--- glibc/Makerules	26 Aug 2007 02:18:03 -0000	1.450
+++ glibc/Makerules	21 Jan 2008 09:41:42 -0000
@@ -993,7 +993,8 @@ $(inst_libdir)/libc.so: $(common-objpfx)
 	 cat $<; \
 	 echo 'GROUP ( $(slibdir)/libc.so$(libc.so-version)' \
 	      '$(libdir)/$(patsubst %,$(libtype.oS),$(libprefix)$(libc-name))'\
-	      ' AS_NEEDED (' $(slibdir)/$(rtld-installed-name) ') )' \
+	      ' AS_NEEDED (' $(slibdir)/$(rtld-installed-name) ') )' ;\
+	 echo 'INPUT ( AS_NEEDED ( -lpthread-stubs ) )' \
 	) > $@.new
 	mv -f $@.new $@
 
Index: sysdeps/mach/hurd/libc-ldscript
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/libc-ldscript,v
retrieving revision 1.2
diff -u -p -r1.2 libc-ldscript
--- glibc/sysdeps/mach/hurd/libc-ldscript	4 Dec 1996 01:41:11 -0000	1.2
+++ glibc/sysdeps/mach/hurd/libc-ldscript	20 Jan 2008 22:19:41 -0000
@@ -4,3 +4,4 @@
    */
 
 GROUP ( libcrt.a libmachuser.a libhurduser.a )
+INPUT ( -lpthread-stubs )
